sklearn/preprocessing/_data.py

```diff
@@ -870,11 +872,12 @@ def partial_fit(self, X, y=None, sample_weight=None):
         self : object
             Fitted scaler.
         """
+        xp, _ = get_namespace(X)
         first_call = not hasattr(self, "n_samples_seen_")
         X = self._validate_data(
             X,
             accept_sparse=("csr", "csc"),
-            dtype=FLOAT_DTYPES,
+            dtype=XP_FLOAT_DTYPES(xp),
```

Comment:
I guess this should be renamed to `xp_float_dtypes`?

Reply:
I think we should rather reuse `sklearn._array_api.supported_float_dtypes(xp)` here.